tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -84,22 +86,19 @@ int RelayHandler::handle_input(ACE_HANDLE)
 
   const auto rd_ptr = buffer->rd_ptr();
 
-  OpenDDS::DCPS::Serializer serializer(buffer.get(), false, OpenDDS::DCPS::Serializer::ALIGN_CDR);
-  OpenDDS::RTPS::Header header;
-  if (!(serializer >> header)) {
-    ACE_ERROR((LM_ERROR, "(%P|%t) %N:%l ERROR: RelayHandler::handle_input failed to deserialize RTPS header\n"));
+  OpenDDS::DCPS::RepoId src_guid;
+  OpenDDS::RTPS::MessageParser mp(*buffer);
+  if (!mp.parseHeader()) {
+    ACE_ERROR((LM_ERROR, "(%P|%t) %N:%l ERROR: RelayHandler::parse_message failed to deserialize RTPS header\n"));
     return 0;
   }
 
-  bool is_beacon_message = buffer->length() == OpenDDS::RTPS::BEACON_MESSAGE_LENGTH &&
-    static_cast<CORBA::Octet>(buffer->rd_ptr()[0]) == OpenDDS::RTPS::BEACON_MSG_ID;
-
-  OpenDDS::DCPS::RepoId src_guid;
+  const auto& header = mp.header();
   std::memcpy(src_guid.guidPrefix, header.guidPrefix, sizeof(OpenDDS::DCPS::GuidPrefix_t));
   src_guid.entityId = OpenDDS::DCPS::ENTITYID_PARTICIPANT;
 
   buffer->rd_ptr(rd_ptr);
-  process_message(remote, ACE_Time_Value().now(), src_guid, buffer.get(), is_beacon_message);
+  process_message(remote, ACE_Time_Value().now(), src_guid, buffer.get());
```

Comment:
@iguessthislldo should this use new time utility?  Time_Point?

Reply:
I'll use MonotonicTimePoint.

Reply:
I'm not as concerned about using the new time types outside the main libraries, but it would be nice. As a warning, `ACE_Time_Value`s require explicit conversions to interact with the new time types, so if just you change one type, most if not all the things the former `ACE_Time_Value` touched will become invalid.